spotify/models/user.py

```diff
@@ -450,6 +450,20 @@ def refresh(self):
         playlist_data = await self.http.create_playlist(self.id, **data)
         return Playlist(self.__client, playlist_data, http=self.http)
 
+    @ensure_http
+    async def follow_playlist(self, playlist, public=True):
+        """follow a playlist
+
+        Parameters
+        ----------
+        playlist : Union[:class:`str`, Playlist]
+            The playlist to modify
+        public : Optional[bool]
+            The public/private status of the playlist.
+            `True` for public, `False` for private.
+        """
+        await self.http.follow_playlist(playlist.id, public)
```

Comment:
This will fail if `playlist` is a string, prefer `str(playlist)` since iirc `Playlist` subclasses `URIBase` and `URIBase` provides a `__str__` which returns the playlist id and in the case where `playlist` is a string, the string will just be copied.

Reply:
Alright I'll update it to: self.http.follow_playlist(self.id, to_id(str(playlist)), public)

Reply:
your http endpoint handler has only two parameters specified, so I have no idea why there are now three arguments provided.

Otherwise, again, good catch I lapsed over mentioning using `to_id` over a uri string :)

Reply:
I have no idea how that self.id got into the code I just pasted!  Its definitely not there in what I'll put up.